deps/libneon/src/ne_openssl.c

```diff
@@ -409,7 +409,7 @@ static ne_ssl_certificate *make_chain(STACK_OF(X509) *chain)
             NE_DEBUG(NE_DBG_SSL, "Cert #%d:", n);
             BIO *mem = BIO_new(BIO_s_mem());
             X509_print(mem, cert->subject);
-            BIO_puts(mem, ""); // force \0
```

Comment:
Hey Petr,

Would you know why this did not work?
The man page mentions: `attempts to write a null terminated string buf [..]`

Reply:
Man page is a bit confusing, because it means you can use BIO_puts to add null terminated string, but null character is not copied in the BIO memory. BIO_write  + "" + size 1 explicitly adds one character from empty string => '\0'

Reply:
Good find! Now, with the `BIO_write` call, the `\0` writing intention is more explicit.